name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/actions/InsertAction.java

```diff
@@ -87,6 +98,38 @@ public Status process(BuySellEntry entry, Account account, Portfolio portfolio)
         // security via the dialog)
         process(entry.getPortfolioTransaction().getSecurity());
 
+        // when importing transactions that have already been generated by an
+        // investment plan, find the existing item and update it
+        if (investmentPlanItem)
+        {
+            DetectDuplicatesAction action = new DetectDuplicatesAction(client);
+            Transaction existingTransaction = null;
+            PortfolioTransaction t = entry.getPortfolioTransaction();
+
+            // search for a match in existing investment plan transactions
+            List<InvestmentPlan> plans = client.getPlans();
+            Iterator<InvestmentPlan> i = plans.stream().filter(p -> p.getSecurity().equals(t.getSecurity())).iterator();
+            while (i.hasNext())
+            {
+                List<Transaction> transactions = i.next().getTransactions();
+                existingTransaction = action.findInvestmentPlanTransaction(t, transactions);
+            }
+
+            // update existingTransaction
+            if (existingTransaction != null)
+            {
+                existingTransaction.setDateTime(t.getDateTime());
+                existingTransaction.setNote(t.getNote());
+                existingTransaction.setShares(t.getShares());
+                existingTransaction.clearUnits();
+                t.getUnits().forEach(existingTransaction::addUnit);
+                
+                if (existingTransaction.getCrossEntry() != null)
+                    existingTransaction.getCrossEntry().getCrossTransaction(existingTransaction).setNote(t.getNote());
+            }
+            return Status.OK_STATUS;
```

Comment:
Maybe the OK_STATUS should only be returned *if* a transaction if updated (i.e. inside the if statement).

If the duplicate is not found (for whatever reasons - parallel updates), then a new transaction is created (in the later code block).

What do you think?

Reply:
Well, that should not happen, but you never know. I adjusted it accordingly.